### PR TITLE
[#64] As a developer, I can run the test for template in CI

### PR DESCRIPTION
**Pull request by luongvo**
https://github.com/nimblehq/flutter_templates/issues/64

## What happened 👀

We don't have any CI for verifying the init script is still working correctly.

## Insight 📝

Create a workflow in CI to verify the template is working correctly:
- Add a step to run the command `make run`
- CI fails if a test fails.
- Trigger the workflow on `push` and `pull request`.
- We need to skip requesting [enquiries](https://pypi.org/project/enquiries/) on CI.

## Proof Of Work 📹

image

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+  push:
+    branches-ignore:
+      - develop
+      - 'release/**'
+      - main
+
+jobs:
+  lint_and_test:
```
Comment (manh-t) [inline .github/workflows/test.yml:14]: I think we should update the job's name to match with the current steps. How about `run_template_script`? 💡

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    name: Template initializing scripts test
+    runs-on: ubuntu-latest
+    environment: staging
```
Comment (manh-t) [inline .github/workflows/test.yml:17]: Do we need to set this environment? 🤔

**Review by manh-t — COMMENTED** [review]
rest LGTM

File: .github/workflows/test.yml (hunk 1 above)
Comment (luongvo) [reply .github/workflows/test.yml:14]: @manh-t `test` is enough I think

File: .github/workflows/test.yml (hunk 2 above)
Comment (luongvo) [reply .github/workflows/test.yml:17]: @manh-t accidentally no, thanks 😄

**Review by manh-t — APPROVED** [review]
LGTM 🚀

**Comment by manh-t** [comment]
@markgravity @doannimble your turn 🙋

File: scripts/setup.py [hunk 3]
```diff
@@ -411,14 +411,18 @@ def clean_up(files: list[str]):
...
+            'snake (snake_case)': 'snake',
+            'pascal (PascalCase)': 'pascal'
+        }
+        choice = enquiries.choose('Choose default json_serializable.field_rename: ', options.keys())
+        project.json_serializable = JsonSerializable(options[choice])
+    else:
+        # Skip enquiries on CI
+        project.json_serializable = JsonSerializable('snake')
 
     print(f"=> 🐢 Staring init {project.new_project_name} with {project.new_package}...")
```
Comment (doannimble) [inline scripts/setup.py:427]: @luongvo typo `Staring` -> `Starting` 🤔

**Review by doannimble — COMMENTED** [review]
Just a small typo mistake, rest lgtm

File: scripts/setup.py (hunk 3 above)
Comment (luongvo) [reply scripts/setup.py]: @doannimble oops, it's not from my PR but fixed 🐰

**Review by doannimble — APPROVED** [review]
Good to go now 🏃

**Pull request merged**